pkg/policy/mapstate.go

```diff
@@ -624,19 +624,16 @@ func (e *mapStateEntry) merge(entry *mapStateEntry) {
 			}
 		}
 
-		// Explicit auth takes precedence over defaulted one.
-		if entry.HasAuthType == ExplicitAuthType {
-			if e.HasAuthType == ExplicitAuthType {
-				// Numerically higher AuthType takes precedence when both are explicitly defined
-				if entry.AuthType > e.AuthType {
-					e.AuthType = entry.AuthType
-				}
-			} else {
-				e.HasAuthType = ExplicitAuthType
+		if entry.HasAuthType == e.HasAuthType {
```

Comment:
Is this a fix for a previous merged commit, or should this be folded in to the PR?

Reply:
I moved this to be the first commit, will post this as a separate PR.

Reply:
On second thought, in 1.16 we only ever support one auth type in datapath (SPIRE), so this is not a bug, but future proofing.